File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        #loop through array, build a hash of what needed as key amd value indes
        #check if what is needed is at current index
        
        needed = {}
        for idx, val  in enumerate(nums):
            if val in needed:
                return [needed[val], idx]
            else:
                needed[target - val] = idx
                
        

File: 1-two-sum/1-two-sum.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
    dictionary = Hash.new
    i = 0
    while i < nums.length
        return dictionary[target-nums[i]], i if dictionary[target-nums[i]]
        dictionary[nums[i]] = i

        i += 1
    end
    []  
end